examples/model_selection/plot_precision_recall.py

```diff
@@ -71,76 +85,146 @@
 X = np.c_[X, random_state.randn(n_samples, 200 * n_features)]
 
+# Limit to the two first classes, and split into training and test
+X_train, X_test, y_train, y_test = train_test_split(X[y < 2], y[y < 2],
+                                                    test_size=.5,
+                                                    random_state=random_state)
+
+# Create a simple classifier
+classifier = svm.LinearSVC(random_state=random_state)
+classifier.fit(X_train, y_train)
+y_score = classifier.decision_function(X_test)
+
+###############################################################################
+# Compute the average precision score
+# ...................................
+from sklearn.metrics import average_precision_score
+average_precision = average_precision_score(y_test, y_score)
+
+print('Average precision-recall score: {0:0.2f}'.format(
+      average_precision))
+
+###############################################################################
+# Plot the Precision-Recall curve
+# ................................
+from sklearn.metrics import precision_recall_curve
+import matplotlib.pyplot as plt
+
+precision, recall, _ = precision_recall_curve(y_test, y_score)
+
+plt.step(recall, precision, color='b', alpha=0.2,
+         where='post')
+plt.fill_between(recall, precision, step='post', alpha=0.2,
+                 color='b')
+
+plt.xlabel('Recall')
+plt.ylabel('Precision')
+plt.ylim([0.0, 1.05])
+plt.xlim([0.0, 1.0])
+plt.title('2-class Precision-Recall curve: AUC={0:0.2f}'.format(
+          average_precision))
+
+###############################################################################
+# In multi-label settings
+# ------------------------
+#
+# Create multli-label data, fit, and predict
+# ...........................................
+#
+# We create a multi-label dataset, to illustrate the precision-recall in
+# multi-label settings
+
+from sklearn.preprocessing import label_binarize
+
+# Use label_binarize to be multi-label like settings
+Y = label_binarize(y, classes=[0, 1, 2])
+n_classes = Y.shape[1]
+
 # Split into training and test
-X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=.5,
+X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=.5,
                                                     random_state=random_state)
 
+# We use OneVsRestClassifier for multi-label prediction
+from sklearn.multiclass import OneVsRestClassifier
 # Run classifier
-classifier = OneVsRestClassifier(svm.SVC(kernel='linear', probability=True,
-                                 random_state=random_state))
-y_score = classifier.fit(X_train, y_train).decision_function(X_test)
+classifier = OneVsRestClassifier(svm.LinearSVC(random_state=random_state))
+classifier.fit(X_train, Y_train)
+y_score = classifier.decision_function(X_test)
 
-# Compute Precision-Recall and plot curve
+
+###############################################################################
+# The precision-Recall score in multi-label settings
```

Comment:
inconsistent capitalization. I'd suggest `precision-recall` everywhere.

I'm very confused now about what "the precision-recall score" is. Is it the same thing as average precision? I've never heard the former before. (just as two distinct scores.) The narrative at the top doesn't make it clear